configs/module.go

```diff
@@ -425,3 +433,27 @@ func (m *Module) mergeFile(file *File) hcl.Diagnostics {
 
 	return diags
 }
+
+// gatherProviderLocalNames is a helper function that returns a map of provider
```

Comment:
I was initially slightly confused by the word "returns" here, since this function doesn't have a return value. Maybe "populates"? This could just be me, though.

Reply:
good call! I would wager that when I wrote that comment it did indeed return something - I've been very bad at updating comments as I iterate on this feature and appreciate the reminders!